### PR TITLE
Translate `guides/upgrading/index.md`

### LINKED ISSUE #119: Translate `/upgrading/index.md`, Ember 4.7
Please assign yourself to the issue or let a comment at the very moment you start the translation.<br><br>File: `guides/release/upgrading/index.md`<br>From Ember: **N/A**<br>To: **4.7**

### PR DESCRIPTION
**Pull request by BlueCutOfficial**
Fixes #119

### CODE REVIEW

File: .local.dic [hunk 1]
```diff
@@ -375,6 +375,7 @@ reader/nombrable
 readme
 README
 readonly
+release/nombrable
```
Comment (MrChocolatine) [inline .local.dic:378]:
```diff
readonly
- release/nombrable
recognizers
...
relayout
+ release/nombrable
reload
```

=>

```
yarn lint:fix:local-dic
```

File: guides/upgrading/index.md [hunk 2]
```diff
@@ -1,39 +1,27 @@
-When someone says they are "upgrading" their Ember app, it could mean one of several things, especially if they say that they are "upgrading to Octane."
-This Guide will help fill in the blanks about how to upgrade your app's version, get access to the latest and greatest features, and form a strategy for using Octane's features in your existing apps.
+Quand des devs disent qu'ils "mettent à jour" leur app Ember, ça peut vouloir dire plusieurs choses, en particulier s'ils parlent de "passer à Octane". Cette page du Guide vous aidera à remplir les blancs sur ce qui concerne la mise à jour la version de votre app, l'accès aux dernières et meilleures fonctionnalités, et à définir une stratégie pour utiliser les fonctionnalités d'Octane dans vos apps existantes.
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:1]:
```suggestion
Quand certains disent qu'ils "mettent à jour" leur app Ember, ça peut vouloir dire plusieurs choses, en particulier s'ils parlent de "passer à Octane". Cette page du Guide vous aidera à remplir les blancs sur ce qui concerne la mise à jour la version de votre app, l'accès aux dernières et meilleures fonctionnalités, et à définir une stratégie pour utiliser les fonctionnalités d'Octane dans vos apps existantes.
```

Nitpicking: I would keep the same line breaks at it makes the reading easier, less compact.

File: guides/upgrading/index.md [hunk 3]
```diff
@@ -1,39 +1,27 @@
...
 
-## Details about recent releases
+## Détail des _releases_ récentes
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:3]: Why keeping `release` while we can use the common `version` in French?

```suggestion
## Détail des récentes versions
```

voire même

```suggestion
## Détail des dernières versions
```

File: guides/upgrading/index.md [hunk 4]
```diff
@@ -1,39 +1,27 @@
...
 
-You can learn about what changed in every release by reading the 
-[official Ember blog](https://blog.emberjs.com).
+Vous apprendrez ce qui a changé dans chaque _release_ (version livrée) en lisant [le blog officiel d'Ember](https://blog.emberjs.com).
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:5]: Je trouve la tournure "Vous apprendrez ce qui a changé" un peu bizarre dans la façon de s'adresser aux gens (alors que c'est du bon français oui).

```suggestion
Pour connaître les changements apportés à chaque _release_ (version livrée), visitez [le blog officiel d'Ember](https://blog.emberjs.com).
```

File: guides/upgrading/index.md [hunk 5]
```diff
@@ -1,39 +1,27 @@
...
-Automated tools will help you make the right changes to `package.json` and other files.
-The point of a minor version bump is that you shouldn't need to change anything in your app when you upgrade - it should all keep working, whether or not you choose to adopt new syntaxes and features.
+Mettons que vous voulez mettre à jour une app de `v3.4` à `v3.8`. Bien que vous ayez entendu parler de nouvelles fonctionnalités, votre objectif principal est de suivre les mises à jour de sécurité. Pour faire ce type de mise à jour, suivez les instructions [ici, dans les Guides CLI](https://cli.emberjs.com/release/basic-use/upgrading/). Ce processus vous assurera que toutes les dépendances de la base de code `ember.js` sont également mises à jour. Des outils automatisés vous aideront à faire les bons changements dans le `package.json` et d'autres fichiers. L'intérêt d'un saut de version mineure est que vous ne devriez pas avoir besoin de changer quoi que ce soit dans votre application lors de la mise à jour&nbsp;; tout devrait continuer à fonctionner, que vous choisissiez ou non d'adopter de nouvelles syntaxes et fonctionnalités.
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:9]:
```suggestion
Mettons que vous voulez mettre à jour une app de `v3.4` à `v3.8`. Bien que vous ayez entendu parler de nouvelles fonctionnalités, votre objectif principal est de suivre les mises à jour de sécurité. Pour faire ce type de mise à jour, suivez les instructions [ici, dans le Guide CLI](https://cli.emberjs.com/release/basic-use/upgrading/). Ce processus vous assurera que toutes les dépendances de la base de code `ember.js` sont également mises à jour. Des outils automatisés vous aideront à faire les bons changements dans le `package.json` et d'autres fichiers. L'intérêt d'un saut de version mineure est que vous ne devriez pas avoir besoin de changer quoi que ce soit dans votre application lors de la mise à jour&nbsp;; tout devrait continuer à fonctionner, que vous choisissiez ou non d'adopter de nouvelles syntaxes et fonctionnalités.
```

Nitpicking: I would keep the same line breaks at it makes the reading easier, less compact.

File: guides/upgrading/index.md [hunk 6]
```diff
@@ -1,39 +1,27 @@
...
 
-## Using new features
+## Utiliser de nouvelles fonctionnalités
 
-Once you have upgraded an app's version, some new features may be available out of the box.
-On the other hand, some features will require that you enable them specifically in your app's configuration, since they may change the app's default behavior.
+Une fois que vous avez mis à jour la version d'une app, de nouvelles fonctionnalités peuvent parfois être disponibles d'emblée. En revanche, d'autres fonctionnalités nécessitent que vous les activiez spécifiquement dans la configuration de votre app, car elles peuvent modifier son comportement par défaut.
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:13]: Personally I would write it this way to insist on the fact these features may or may not change the behaviour.
I found the original sentence more like "they can and they will".

```suggestion
Une fois que vous avez mis à jour la version d'une app, de nouvelles fonctionnalités prêtes à l'emploi peuvent parfois être disponibles. En revanche, d'autres fonctionnalités nécessitent que vous les activiez spécifiquement dans la configuration de votre app, car il est possible qu'elles puissent modifier le comportement par défaut de votre app.
```

Nitpicking: I would keep the same line breaks at it makes the reading easier, less compact.

File: guides/upgrading/index.md [hunk 7]
```diff
@@ -1,39 +1,27 @@
...
-In many cases, codemods will be available to help you make syntax-related updates.
-A codemod is a tool that rewrites your existing code into a new syntax.
-When they are available, they can save a lot of time that you would spend making edits by hand.
+<!-- spell ignore -->
+La meilleure manière de découvrir les nouvelles fonctionnalités est de lire les [articles de blog sur les _releases_](https://blog.emberjs.com/tags/releases.html). Une nouvelle fonctionnalité qui requiert que vous l'activiez est appelée fonctionnalité optionnelle (_optional feature_). Suivez [le guide des fonctionnalités optionnelles](../configuring-ember/optional-features/) pour savoir lesquelles sont disponibles dans la version d'Ember de votre app, et comment les activer. Dans de nombreux cas, des _codemods_ seront disponibles pour vous aider à faire des mises à jour liées à la syntaxe. Un _codemod_ est un outil qui réécrit votre code existant dans une nouvelle syntaxe. Lorsqu'ils sont disponibles, ils peuvent vous épargner beaucoup de temps passé à faire des modifications à la main.
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:16]:
```suggestion
La meilleure manière de découvrir les nouvelles fonctionnalités est de lire les [articles de blog sur les _releases_](https://blog.emberjs.com/tags/releases.html). Une nouvelle fonctionnalité qui requiert que vous l'activiez est appelée fonctionnalité optionnelle (_optional feature_). Suivez [le guide des fonctionnalités optionnelles](../configuring-ember/optional-features/) pour savoir lesquelles sont disponibles dans la version d'Ember de votre app, et comment les activer. Dans de nombreux cas, des _codemods_ seront disponibles pour vous aider à faire des mises à jour liées à la syntaxe. Un _codemod_ est un outil qui réécrit votre code actuel dans une nouvelle syntaxe. Lorsqu'ils sont disponibles, ils peuvent vous épargner beaucoup de temps passé à faire des modifications à la main.
```

File: guides/upgrading/index.md [hunk 8]
```diff
@@ -1,39 +1,27 @@
...
-Sometimes, they will be deprecation warnings caused by code in your app, and other times, they may be caused by an addon.
+Si une API que vous utiliser doit être supprimée dans la prochaine version majeure d'Ember, vous verrez un message de dépréciation (_deprecation warning_) dans la console de développement. Ces messages pourront parfois être causés par le code de votre app, et d'autres fois, par celui d'un addon.
 
-For more guidance on what to do with deprecations, visit [Handling Deprecations](../configuring-ember/handling-deprecations/), check out the Ember Inspector [tools for deprecations](../ember-inspector/deprecations/), or read about the specifics in the [Deprecations Guides](https://deprecations.emberjs.com/).
+Pour en savoir plus sur la manière dont gérer les dépréciations, lisez la page [Gérer les dépréciations](../configuring-ember/handling-deprecations/), jetez un œil [aux outils de gestion des dépréciations](../ember-inspector/deprecations/) de l'Ember Inspector, ou regardez les [_Deprecations Guides_](https://deprecations.emberjs.com/).
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:22]:
```suggestion
Pour en savoir plus sur la manière de gérer les dépréciations, lisez la page [Gérer les dépréciations](../configuring-ember/handling-deprecations/), jetez un œil [aux outils de gestion des dépréciations](../ember-inspector/deprecations/) de l'Ember Inspector, ou regardez les [_Deprecations Guides_](https://deprecations.emberjs.com/).
```

File: guides/upgrading/index.md [hunk 9]
```diff
@@ -1,39 +1,27 @@
...
 
-## Upgrading to Octane
 
-Octane was a big shift in Ember's syntax, features, and mental models! If you are in the process of upgrading an existing app to use Octane patterns, check out our dedicated [Octane Update Guide](./current-edition/).
+## Mettre à jour vers Octane
+
+Octane a été un grand pas en avant dans la syntaxe, les fonctionnalités, et les modèles mentaux d'Ember&nbsp;! Si vous êtes dans le processus de mettre à jour une app existante vers Octane, jetez un œil à la page dédiée [Guide de Mise à Jour vers Octane](./current-edition/).
```
Comment (MrChocolatine) [inline guides/upgrading/index.md:27]: I would keep the preterit.

```suggestion
Octane fut un grand pas en avant dans la syntaxe, les fonctionnalités, et les modèles mentaux d'Ember&nbsp;! Si vous êtes en train de mettre à jour une app existante vers Octane, jetez un œil à la page dédiée [Guide de Mise à Jour vers Octane](./current-edition/).
```

File: guides/upgrading/index.md (hunk 5 above)
Comment (BlueCutOfficial) [reply guides/upgrading/index.md:9]: To be honest their pseudo-line breaks that are not proper line-breaks get me mad and make my reading harder. I prefer the markdown to represent clearly what I see on screen. It's a question of preference really.

**Review by MrChocolatine — APPROVED** [review]

**Pull request merged**